### PR TITLE
Build system overhaul.

### PR DESCRIPTION
**Pull request by hestela**
1. Switched from a bunch of mods to a bunch of crates.
2. Moved files out of psm mod into their own crates.
3. Added empty ptl and ipath crates for thier C psm counterparts.

No functional changes in this patch.

@acmcarther

### CODE REVIEW

File: ep/src/lib.rs [hunk 1]
```diff
@@ -1,13 +1,15 @@
-use psm::epid::*;
-use psm::error::*;
-use psm::consts::*;
+extern crate error;
+use error::*;
```
Comment (acmcarther) [inline ep/src/lib.rs]: Are we using all of the exports from error? I might avoid importing the whole universe unless we're using tons and tons of em.

File: ep/src/lib.rs [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
 use std::rc::Rc;
+use self::consts::*;
```
Comment (acmcarther) [inline ep/src/lib.rs:5]: Weird. What does this do?

File: error/src/lib.rs [hunk 3]
```diff
@@ -4,7 +4,7 @@ pub enum Error_type {
     PSM_OK_NO_PROGRESS = 1,
     PSM_PARAM_ERR = 3,
     PSM_NO_MEMORY = 4,
-    PSM_INIT_NOT_INIT = 5, 
+    PSM_INIT_NOT_INIT = 5,
```
Comment (acmcarther) [inline error/src/lib.rs:7]: Ooh, weird issue 2. Do we need explicit conversions here for backwards compatibility?

**Comment by acmcarther** [comment]
Nice stuff pulling the modules into crates. Can't wait to check it out when they're more fully fleshed out.

:shipit:

File: error/src/lib.rs (hunk 3 above)
Comment (hestela) [inline error/src/lib.rs:7]: In the C code, it is a plain enum as well, they mainly did stuff like  `if(err_code == PSM_OK)`. I just copied all of their error codes here.

Comment (acmcarther) [inline error/src/lib.rs:7]: Ah yeah. I'd just be on the lookout to see if we need the explicit numbering. In rust, you almost certainly won't but there might be some kind of voodoo significance of these digits. Again, thats probably a code smell but whatevs.

I guess yeah roll with it for now.

File: ep/src/lib.rs (hunk 1 above)
Comment (hestela) [inline ep/src/lib.rs]: In this instance im using 2/3 things in the error crate, but most of the time fns will only need to return an error code. So ill start importing what i need at the moment.

Comment (acmcarther) [inline ep/src/lib.rs]: :balloon:

File: ep/src/lib.rs (hunk 2 above)
Comment (hestela) [inline ep/src/lib.rs:5]: I'm not sure if this is correct, but it made it compile. This is letting me use the stuff from the consts.rs in the same crate.

**Comment by acmcarther** [comment]
What have you done!?!

**Comment by hestela** [comment]
Forgot that github doesnt work like gerrit, thx work.

**Comment by acmcarther** [comment]
You should unwreck this with a force push or something

**Comment by hestela** [comment]
☐ rekt
☑ not rekt

**Comment by acmcarther** [comment]
DELETE branch

**Pull request merged**